karpenter-operator/controllers/karpenter/assets/adjust-cel.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+# remove amiSelectorTerms, securityGroupSelectorTerms, subnetSelectorTerms required validation
+yq eval '.spec.versions[0].schema.openAPIV3Schema.properties.spec.required = []' -i ${SCRIPT_DIR}/karpenter.k8s.aws_ec2nodeclasses.yaml
+
+# since miSelectorTerms is no longer required, top level validations need to be adjusted accordingly to check for the existence of the field before accessing it.
+yq eval '.spec.versions[0].schema.openAPIV3Schema.properties.spec.x-kubernetes-validations = [
+    {"message": "must specify exactly one of ['role', 'instanceProfile']", "rule": "(has(self.role) && !has(self.instanceProfile)) || (!has(self.role) && has(self.instanceProfile))"},
+    {"message": "changing from 'instanceProfile' to 'role' is not supported. You must delete and recreate this node class if you want to change this.", "rule": "(has(oldSelf.role) && has(self.role)) || (has(oldSelf.instanceProfile) && has(self.instanceProfile))"},
+    {"message": "if set, amiFamily must be 'AL2' or 'Custom' when using an AL2 alias", "rule": "!has(self.amiFamily) || (has(self.amiSelectorTerms) && self.amiSelectorTerms.exists(x, has(x.alias) && x.alias.find(\"^[^@]+\") == \"al2\") ? (self.amiFamily == \"Custom\" || self.amiFamily == \"AL2\") : true)"},
```

Comment:
since we are here, we can actually change this to enforce amiFamily == custom || empty and that amiSelectorTerms fields must be empty other than .id

Reply:
done.